### PR TITLE
feat(components): add proposal detail page structure

### PR DESCRIPTION
**Pull request by jennieramida**

### CODE REVIEW

File: src/lib/pages/proposal-details/index.tsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+import PageContainer from "lib/components/PageContainer";
+import { EmptyState } from "lib/components/state";
+
+import { ProposalDetailBody } from "./components/ProposalDetailsBody";
+import { ProposalTop } from "./components/ProposalTop";
+
+export const ProposalDetail = () => {
+  const router = useRouter();
+  const proposalId = parseInt(router.query.id as string, 10);
```
Comment (evilpeach) [inline src/lib/pages/proposal-details/index.tsx:11]: Let's use zod. See the other details page as example

**Review by evilpeach — CHANGES_REQUESTED** [review]

File: src/lib/pages/proposal-details/components/ProposalDetailsBody.tsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+import type { ProposalDetailsQueryParams } from "../type";
+import { TabIndex } from "../type";
+import { useInternalNavigate } from "lib/app-provider";
+import { CustomTab } from "lib/components/CustomTab";
+
+import { ProposalOverview } from "./ProposalOverview";
+import { VoteDetail } from "./VoteDetail";
+
+export const ProposalDetailBody = ({
```
Comment (songwongtp) [inline src/lib/pages/proposal-details/components/ProposalDetailsBody.tsx:13]: ProposalDetailsBody

File: src/lib/pages/proposal-details/components/ProposalDetailsBody.tsx [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+          id,
+          tab: TabIndex.Overview,
+        },
+        options: {
+          shallow: true,
+        },
+      });
+    }
+  }, [router.isReady, tab, navigate, id]);
+
```
Comment (songwongtp) [inline src/lib/pages/proposal-details/components/ProposalDetailsBody.tsx:53]: handle invalid proposal in here

File: src/lib/pages/proposal-details/index.tsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+}: ProposalDetailsQueryParams) => {
+  // TODO Mock up
+  if (id === 999) return <InvalidState title="Proposal does not exist" />;
+  return (
+    <>
+      <ProposalTop id={id} />
+      <ProposalDetailBody id={id} tab={tab} voteTab={voteTab} />
+    </>
+  );
+};
```
Comment (songwongtp) [inline src/lib/pages/proposal-details/index.tsx:24]: remove

File: src/lib/pages/proposal-details/index.tsx [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+export const ProposalDetails = () => {
```
Comment (songwongtp) [inline src/lib/pages/proposal-details/index.tsx:25]: for consistency, remove export

File: src/lib/pages/proposal-details/index.tsx [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+  return (
+    <PageContainer>
+      {!validated.success ? (
+        <InvalidState title="Proposal does not euuuxist" />
+      ) : (
+        <ProposalDetailsBody {...validated.data} />
+      )}
+    </PageContainer>
+  );
+};
```
Comment (songwongtp) [inline src/lib/pages/proposal-details/index.tsx:39]: `export default ProposalDetails`

File: src/lib/pages/proposal-details/components/ValidatorVotesTable.tsx [hunk 7]
```diff
@@ -0,0 +1,41 @@
+import type { FlexProps } from "@chakra-ui/react";
+import { Button, Flex } from "@chakra-ui/react";
+
+import { useMobile } from "lib/app-provider";
+import { CustomIcon } from "lib/components/icon";
+import { TableTitle } from "lib/components/table";
+
+interface ValidatorVotesTableProps extends FlexProps {
+  onBack: () => void;
+}
```
Comment (songwongtp) [inline src/lib/pages/proposal-details/components/ValidatorVotesTable.tsx:10]: new line

File: src/lib/pages/proposal-details/index.tsx [hunk 8]
```diff
@@ -0,0 +1,39 @@
+import { useRouter } from "next/router";
+
+import PageContainer from "lib/components/PageContainer";
+import { InvalidState } from "lib/components/state";
+
+import { ProposalDetailBody } from "./components/ProposalDetailsBody";
```
Comment (songwongtp) [inline src/lib/pages/proposal-details/index.tsx:6]: add `index.ts` in the component dir with
```
export * from "./ProposalDetailsBody";
export * from "./ProposalTop";
```
then import from just `./components` instead

**Review by songwongtp — CHANGES_REQUESTED** [review]

File: src/lib/pages/proposal-details/components/ProposalDetailsBody.tsx (hunk 3 above)
Comment (jennieramida) [reply src/lib/pages/proposal-details/components/ProposalDetailsBody.tsx:53]: already handle in index page

File: src/lib/components/MobileGuard.tsx [hunk 9]
```diff
@@ -24,6 +24,7 @@ export const MobileGuard = ({ children }: MobileGuardProps) => {
     pathName.includes(`/network-overview`) ||
     pathName.includes(`/dev-home`) ||
     pathName.includes(`/404`) ||
+    pathName.includes(`/proposals`) ||
```
Comment (evilpeach) [inline src/lib/components/MobileGuard.tsx:27]: I think we should update as following: 
```tsx
export const MobileGuard = ({ children }: MobileGuardProps) => {
  const router = useRouter();
  const pathName = router.asPath;
  const isMobile = useMobile();
  const isResponsive =
    pathName.includes(`/accounts`) ||
    pathName.includes(`/txs`) ||
    pathName.includes(`/blocks`) ||
    pathName.includes(`/projects`) ||
    pathName.includes(`/codes`) ||
    pathName.includes(`/nft-collections`) ||
    pathName.includes(`/query`) ||
    pathName.includes(`/404`) ||
    pathName.includes(`/proposals`) ||
    // wasm
    pathName.includes(`/contracts/`) ||
    // move
    pathName.includes(`/modules`);

  if (isResponsive && isMobile) return <>{children}</>;
  if (!isResponsive && isMobile) return <NoMobile />;
  return <>{children}</>;
};
```

**Review by songwongtp — APPROVED** [review]

**Review by evilpeach — APPROVED** [review]

**Pull request merged**